### DISCUSSION TITLE
datadog exporter

### DISCUSSION THREAD

**Discussion by hamidp555** [open-telemetry/opentelemetry-collector-contrib#21407] [Q&A]
I am trying to secure a connection between otel collector datadog exporter and an nginx reverse proxy. I have setup nginx with self signed certificates. However, in my collector I get `proxyconnect tcp: x509: certificate signed by unknown authority` error` , and in my nginx i get this error ` SSL_do_handshake() failed (SSL: error:14094412:SSL routines:ssl3_read_bytes:sslv3 alert bad certificate:SSL alert number 42) while SSL handshaking, client`

Attempting to solve this issue, I was thinking to provide ca.crt of the nginx proxy in datadog exporter under tls: ca_file: <CA_FILE> but datadog doesn't allow this property to be set.

I appreciate if anyone has ideas of fixing this error.

**Comment by atoulme** [comment]
Your nginx endpoint is missing a certificate, please fix your installation with let's encrypt or a SSL certificate. You can also configure the collector to disable verifying the SSL certificate. See `insecure_skip_verify` https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/datadogexporter/config.go#LL333C41-L333C61

**Reply by sreejithav** [reply]
i am getting this error even if i set insecure_skip_verify=true ...  _"debug   datadogexporter@v0.70.0/zaplogger.go:39 Retrying to flush trace payload; error: Post "https://trace.agent.us5.datadoghq.com/api/v0.2/traces": x509: certificate signed by unknown authority    {"kind": "exporter", "data_type": "metrics", "name": "datadog"}_ . looks like insecure_skip_verify works only for logs not for traces.

my setup is simple... i am using the demo app from opentelemetry and with a datadog exporter.

appreciate any help here.
